parachains/runtimes/assets/asset-hub-westend/src/lib.rs

```diff
@@ -1360,3 +1376,113 @@ cumulus_pallet_parachain_system::register_validate_block! {
 	BlockExecutor = cumulus_pallet_aura_ext::BlockExecutor::<Runtime, Executive>,
 	CheckInherents = CheckInherents,
 }
+
+pub mod migrations {
+	use super::*;
+	use frame_support::{
+		pallet_prelude::Get,
+		traits::{
+			fungible::{Inspect as InspectFungible, Mutate as MutateFungible},
+			fungibles::{Inspect, Mutate},
+			tokens::Preservation,
+			OnRuntimeUpgrade, OriginTrait,
+		},
+	};
+	use parachains_common::impls::AccountIdOf;
+	use sp_runtime::traits::StaticLookup;
+	use xcm::latest::prelude::*;
+
+	/// Temporary migration because of bug with native asset, it can be removed once applied on `AssetHubWestend`.
+	/// Migrates pools with `MultiLocation { parents: 0, interior: Here }` to `MultiLocation { parents: 1, interior: Here }`
+	pub struct NativeAssetParents0ToParents1Migration<T>(sp_std::marker::PhantomData<T>);
+	impl<
+			T: pallet_asset_conversion::Config<
+				MultiAssetId = Box<MultiLocation>,
+				AssetId = MultiLocation,
+			>,
+		> OnRuntimeUpgrade for NativeAssetParents0ToParents1Migration<T>
+	where
+		<T as pallet_asset_conversion::Config>::PoolAssetId: Into<u32>,
+		AccountIdOf<Runtime>: Into<[u8; 32]>,
+		<T as frame_system::Config>::AccountId:
+			Into<<<T as frame_system::Config>::RuntimeOrigin as OriginTrait>::AccountId>,
+		<<T as frame_system::Config>::Lookup as StaticLookup>::Source:
+			From<<T as frame_system::Config>::AccountId>,
+		sp_runtime::AccountId32: From<<T as frame_system::Config>::AccountId>,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let invalid_native_asset = MultiLocation { parents: 0, interior: Here };
+			let valid_native_asset = WestendLocation::get();
+
+			let mut reads = 1;
+			let mut writes = 0;
+
+			// migrate pools with invalid native asset
+			let pools = pallet_asset_conversion::Pools::<T>::iter().collect::<Vec<_>>();
+			for (old_pool_id, pool_info) in pools {
+				let old_pool_account =
+					pallet_asset_conversion::Pallet::<T>::get_pool_account(&old_pool_id);
+				reads += 1;
```

Comment:
Probably fine, but paranoia.
```suggestion
				reads.saturating_accrue(1);
```

Reply:
ok, thx, fixed everywhere